Update CHANGELOG for new AWS modules

willthames opened the pull request:
##### SUMMARY
Reflect new AWS modules in changelog



##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
CHANGELOG

##### ANSIBLE VERSION
```
ansible 2.4.0 (devel d1652aecf0) last updated 2017/06/30 14:21:05 (GMT +1000)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/will/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/will/src/ansible/lib/ansible
  executable location = /home/will/src/ansible/bin/ansible
  python version = 2.7.13 (default, May 10 2017, 20:04:28) [GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]
```


##### ADDITIONAL INFORMATION
Used:

```
git diff --diff-filter A --name-only --oneline stable-2.3 -- \
  lib/ansible/modules/cloud/amazon/  | xargs -n1 -I{} basename {} .py \
  | xargs -n1 echo '  *'
```

ryansb reviewed (approved):
Looks good to me. Technically the SNS module was moved, so it's not "new" entirely.

Commit:
Update CHANGELOG for new AWS modules

Used:

```
git diff --diff-filter A --name-only --oneline stable-2.3 -- \
  lib/ansible/modules/cloud/amazon/  | xargs -n1 -I{} basename {} .py \
  | xargs -n1 echo '  *'
```

willthames commented:
I've removed sns, and will merge once tests pass (which they should, trivially)
